lib/spack/spack/solver/asp.py

```diff
@@ -3188,9 +3138,8 @@ def setup(
         self.gen.h1("Variant Values defined in specs")
         self.define_variant_values()
 
-        if WITH_RUNTIME:
-            self.gen.h1("Runtimes")
-            self.define_runtime_constraints()
+        self.gen.h1("Runtimes")
```

Comment:
We should eventually call this something else -- e.g. injected constraints (i.e. injected from packages) as "runtime constraints" is more specific than what this really enables. Can be for after the PR though -- it is already more general than it was before.

Reply:
Yeah, I agree. We should probably come up with different names for "runtime" entities, and a generalization of the concept of "runtimes" in a following PR.  

We also need to consider `hip` as a use case when generalizing the model, since it introduces a runtime with a huge DAG, whereas up to now we kind of assume runtimes are relatively small, and extracted from a "compiler" package.

Reply:
DAG may be huge but that's why this is nice ... it integrates with the resolver, so it's not like we have to add everything, right?  Anyway -- for later.